### PR TITLE
Replace OC dialog with NcDialog

### PR DESCRIPTION
**Pull request by DorraJaouad**
### 🖼️ Screenshots / Screencasts

| 🏚️ Before | 🏡 After |
|------------|----------|
| image  |image     |
| image | image|
| image | image

### 🚧 Tasks

- [x] Code review
- [x] Visual review

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [x] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

**Comment by nickvergessen** [comment]
Ah, I missed this is still open. I bumped dialogs in #10821  So make sure to rebase and retest :)

**Comment by DorraJaouad** [comment]
Before I move further, we should migrate the duplicate session dialog to NcDialog too, right? 

https://github.com/nextcloud/spreed/blob/main/src/store/participantsStore.js#L790-L825

It will be replaced by an event emission from the store and listened by the main components calling `'joinConversation'`  : 

`App.vue`
`FilesSidebarTabApp.vue`
`PublicShareAuthSidebar.vue`
`PublicShareSidebar.vue`

For the others, like `LeftSidebar`'s function call, we can use the `NcDialog` in App component instead

**Comment by nickvergessen** [comment]
Lets do that in a different PR. It's a more problematic thing to check and test than the currently handled deletions.

File: src/components/ConversationSettings/DangerZone.vue [hunk 1]
```diff
@@ -56,9 +70,23 @@
...
+				<NcDialog class="danger-zone__dialog"
+					:open.sync="isDeleteChatDialogOpen"
+					:name="t('spreed','Delete all chat messages')"
+					:message="deleteChatDialogMessage"
+					:container="container">
+					<template #actions>
+						<NcButton type="tertiary" @click="toggleShowDeleteChatDialog">
+							{{ t('spreed', 'No') }}
+						</NcButton>
+						<NcButton type="primary" @click="clearChatHistory">
```
Comment (nickvergessen) [inline src/components/ConversationSettings/DangerZone.vue:85]:
```suggestion
						<NcButton type="error" @click="clearChatHistory">
```

File: src/components/ConversationSettings/DangerZone.vue [hunk 2]
```diff
@@ -44,9 +44,23 @@
...
+				<NcDialog class="danger-zone__dialog"
+					:open.sync="isDeleteConversationDialogOpen"
+					:name="t('spreed','Delete Conversation')"
+					:message="deleteConversationDialogMessage"
+					:container="container">
+					<template #actions>
+						<NcButton type="tertiary" @click="toggleShowDeleteConversationDialog">
+							{{ t('spreed', 'No') }}
+						</NcButton>
+						<NcButton type="primary" @click="deleteConversation">
```
Comment (nickvergessen) [inline src/components/ConversationSettings/DangerZone.vue:59]:
```suggestion
						<NcButton type="error" @click="deleteConversation">
```

File: src/components/LeftSidebar/ConversationsList/Conversation.vue [hunk 3]
```diff
@@ -20,102 +20,119 @@
...
+		<!-- confirmation required to delete conversation -->
+		<NcDialog :open.sync="isDialogOpen"
+			:name="t('spreed','Delete Conversation')"
+			:message="dialogMessage"
+			:container="container">
+			<template #actions>
+				<NcButton type="tertiary" @click="closeDialog">
+					{{ t('spreed', 'No') }}
+				</NcButton>
+				<NcButton type="primary" @click="deleteConversation">
```
Comment (nickvergessen) [inline src/components/LeftSidebar/ConversationsList/Conversation.vue:125]:
```suggestion
				<NcButton type="error" @click="deleteConversation">
```

**Review by nickvergessen — APPROVED** [review]
I would use error in all cases like you used already for the breakout room deletion

**Pull request merged**